### PR TITLE
fix verify-attached-bugs cli

### PR DESCRIPTION
**Pull request by Ximinhan**
the issue is due to the API init too early, and is_first_fix check get bugs one by one which takes over 30 mins for RC.
In this pr:
- replace get_bug with get_bugs in is_first_fix check
- use Erratum replace some get advisory API
- move login check to request function for cve_package_exclusions API

test build: https://saml.buildvm.hosts.prod.psi.bos.redhat.com:8888/job/hack/job/ximhan/job/build%252Foperator-sdk_sync/52/console

### CODE REVIEW

File: elliottlib/errata_async.py [hunk 1]
```diff
@@ -34,9 +34,9 @@ async def login(self):
     async def _make_request(self, method: str, path: str, parse_json: bool = True, **kwargs) -> Union[Dict, bytes]:
         if "headers" not in kwargs:
             kwargs["headers"] = self._headers
-        async with aiohttp.ClientSession(connector=aiohttp.TCPConnector(limit=32, force_close=True)) as session:
+        await self.login()
```
Comment (thegreyd) [inline elliottlib/errata_async.py:37]: how does this work? wouldn't it be redundant to attempt to login with every request?

File: elliottlib/errata_async.py [hunk 2]
```diff
@@ -34,9 +34,9 @@ async def login(self):
...
+        async with aiohttp.ClientSession(connector=aiohttp.TCPConnector(limit=32, force_close=True), timeout=aiohttp.ClientTimeout(total=60 * 60)) as session:
             async with session.request(method, self._errata_url + path, **kwargs) as resp:
-                resp.raise_for_status()
```
Comment (thegreyd) [inline elliottlib/errata_async.py:39]: why remove this?

File: elliottlib/errata_async.py (hunk 1 above)
Comment (Ximinhan) [reply elliottlib/errata_async.py:37]: the login method is to make sure the SecurityContext is set for the request
and actually in this version only cve_package_exclusion is using this, only four requests will be called

File: elliottlib/bzutil.py [hunk 3]
```diff
@@ -1115,17 +1115,10 @@ def is_first_fix_any(bugtracker, flaw_bug, current_target_release):
 
     # filter tracker bugs by OCP product
     tracker_bugs = []
-    for tracker_id in tracker_ids:
-        try:
-            b = bugtracker.get_bug(tracker_id)
```
Comment (Ximinhan) [inline elliottlib/bzutil.py:1120]: get_bug in sequential execution will takes over 30 mins

File: elliottlib/cli/verify_attached_bugs_cli.py [hunk 4]
```diff
@@ -213,8 +214,8 @@ async def _verify_attached_flaws_for(self, advisory_id: int, attached_trackers:
                            "You need to drop those flaw bugs or attach corresponding tracker bugs.")
 
         # Check if advisory is of the expected type
-        advisory_info = await self.errata_api.get_advisory(advisory_id)
```
Comment (Ximinhan) [inline elliottlib/cli/verify_attached_bugs_cli.py:216]: get advisory can use Erratum

File: elliottlib/cli/verify_attached_bugs_cli.py [hunk 5]
```diff
@@ -178,6 +177,8 @@ async def _verify_attached_flaws_for(self, advisory_id: int, attached_trackers:
         # Retrieve flaw bugs for attached_tracker_bugs
         tracker_flaws, flaw_id_bugs = BugTracker.get_corresponding_flaw_bugs(attached_trackers,
                                                                              self.runtime.bug_trackers('bugzilla'))
+        current_cve_package_exclusions = await AsyncErrataUtils.get_advisory_cve_package_exclusions(self.errata_api, advisory_id)
```
Comment (Ximinhan) [inline elliottlib/cli/verify_attached_bugs_cli.py:180]: start the async call early to get the necessary content before is_first_fix_any function call

File: elliottlib/cli/find_bugs_sweep_cli.py [hunk 6]
```diff
@@ -264,7 +264,7 @@ def categorize_bugs_by_type(bugs: List[Bug], advisory_id_map: Dict[str, int], ma
 
     warning_bug = [b.id for b in non_tracker_bugs if b.is_cve_in_summary()]
     if warning_bug:
-        logger.warn(f"Bug {warning_bug} has CVE number in summary but does not have tracker keywords")
```
Comment (Ximinhan) [inline elliottlib/cli/find_bugs_sweep_cli.py:267]: fix ut warning
```
..........................E...................../mnt/workspace/jenkins/working/art-tools_elliott_PR-477/elliottlib/cli/find_bugs_sweep_cli.py:267: DeprecationWarning: The 'warn' method is deprecated, use 'warning' instead
  logger.warn(f"Bug {warning_bug} has CVE number in summary but does not have tracker keywords")
```

File: elliottlib/errata_async.py (hunk 2 above)
Comment (Ximinhan) [reply elliottlib/errata_async.py:39]: I tested that status 500 or 401 will failed the command and raise the error anyway whether have this line or not

File: elliottlib/cli/verify_attached_bugs_cli.py [hunk 7]
```diff
@@ -48,7 +48,6 @@ async def verify_attached_bugs_cli(runtime: Runtime, verify_bug_status: bool, ad
 async def verify_attached_bugs(runtime: Runtime, verify_bug_status: bool, advisory_id_map: Dict[str, int], verify_flaws:
                                bool, no_verify_blocking_bugs: bool):
     validator = BugValidator(runtime, output="text")
-    await validator.errata_api.login()
```
Comment (Ximinhan) [inline elliottlib/cli/verify_attached_bugs_cli.py:51]: what login method does is add gssapi.SecurityContext and Authorization header, just need to make sure they are set before actually making the request, set them earlier doesn't help much

File: elliottlib/cli/verify_attached_bugs_cli.py (hunk 4 above)
Comment (Ximinhan) [reply elliottlib/cli/verify_attached_bugs_cli.py:216]: so no need to decode json and Erratum is very reliable, and after this only cve_package_exclusions API which not implement in Erratum will called by request

File: elliottlib/cli/verify_attached_bugs_cli.py (hunk 5 above)
Comment (Ximinhan) [reply elliottlib/cli/verify_attached_bugs_cli.py:180]: even is_first_fix_any get bugs in batch will also take 8 mins for RC

File: elliottlib/bzutil.py [hunk 8]
```diff
@@ -1115,17 +1115,10 @@ def is_first_fix_any(bugtracker, flaw_bug, current_target_release):
...
-        except Exception as e:
-            # first-fix tracker bug might be not visible but not break here
-            if "not authorized" in e:
```
Comment (Ximinhan) [inline elliottlib/bzutil.py:1123]: at least for current RC, I didn't see not authorized error or other bug can't get error

**Pull request closed without merging**